File: src/handler.dart
import 'dart:convert';
import 'dart:io';
import 'package:http/http.dart' as http;

void handler() async {
  var response = await get(
    url: "https://api.orats.io/datav2/snapshot/strikes15",
    params: {
      "token": "{YOUR_ORATS_KEY}",
      "tradeDate": "202008201130",
    },
  );
  print(response);
}

Future<dynamic> get({
  required String url,
  Map<String, dynamic>? params = const {},
  Map<String, String>? headers = const {},
  int timeout = 10,
  bool DEBUG = false,
}) async {
  var uri = Uri.parse(url).replace(queryParameters: params);
  var request = http.Request('GET', uri)..headers.addAll(headers ?? {});

  try {
    http.StreamedResponse response = await request.send().timeout(
      Duration(seconds: timeout),
      onTimeout: () {
        return http.StreamedResponse(
          Stream.value(utf8.encode('Request Time Out')),
          408,
        );
      },
    );
    int status = response.statusCode;
    var data = await response.stream.bytesToString();
    
    if (DEBUG) {
      print("[$status] $uri");
      print("- " * 25);
      print("$headers");
      print("- " * 25);
      print(response.headers);
      print("*" * 100);
    }

    if ((response.headers["content-type"] ?? "").contains("application/json")) {
      return jsonDecode(data);
    } else {
      return data;
    }
    
  } catch (e) {
    print('Error: $e');
    return {"error": "$e"};
  }
}

File: src/handler.mjs
// esm
import pl from 'nodejs-polars';
import fetch from 'node-fetch';

// nodejs-polars-linux-x64-gnu
export const handler = async (event) => {
    // TODO implement

    var data = await get({
        url: "https://api.orats.io/datav2/snapshot/strikes15",
        params: {
          "token": "{YOUR_ORATS_KEY}",
          "tradeDate": "202008201130",
        },
    })
    
    const df = pl.DataFrame(flattenRecords(data));
    console.log(df);
  
    df.writeJSON('__data__/output.json')
    df.writeCSV('__data__/output.csv')
    df.writeParquet('__data__/output.parquet')

    const response = {
      statusCode: 200,
      body: JSON.stringify('Hello from Lambda!'),
    };

    return response;
  };


async function get({ url, params = {}, headers = {}, timeout = 60, DEBUG = false}) {
  console.log("hello")
  const queryString = Object.keys(params).map(key => `${key}=${params[key]}`).join('&');
  const fullUrl = `${url}?${queryString}`;

  if (DEBUG) {
    console.log(fullUrl)
    console.log(headers)
  }

  return fetch(fullUrl, {
    method: 'GET',
    headers: headers,
  })
    .then(response => {        
      if (response.headers.get('Content-Type').includes("application/json")) { 
        return response.json() 
      } 
      else { return response.text() }
    })
    .catch(error => {
      if (DEBUG) console.log(error)
      return {"error": error }
    });
}

function flattenRecords(records) {
  return records.map((record) => {
    const flattenedRecord = {};
    flattenObject(record, flattenedRecord, '');
    return flattenedRecord;
  });

  function flattenObject(obj, result, prefix) {
    for (const key in obj) {
      if (typeof obj[key] === 'object' && obj[key] !== null) {
        flattenObject(obj[key], result, prefix + key + '.');
      } else {
        result[prefix + key] = obj[key];
      }
    }
  }
}
